### PR TITLE
[iOS] Refactor MediaState

### PR DESCRIPTION
**Pull request by myihsan**
## Overview (Required)

- Refactor `MediaState` to an enumeration
- Refactor `FeedContents` to `FeedContent`
- Use `FeedContent` to update `MediaListState`

### CODE REVIEW

File: ios/Sources/MediaFeature/MediaScreen.swift [hunk 1]
```diff
@@ -42,11 +43,17 @@ public struct MediaScreen: View {
...
-                then: MediaListView.init(store:),
-                else: { ProgressView().onAppear { viewStore.send(.progressViewAppeared) } }
-            )
+            SwitchStore(store) {
+                CaseLet(
+                    state: /MediaState.initialized,
+                    action: MediaAction.init(action:),
+                    then: MediaListView.init(store:)
+                )
+                Default {
```
[the remark this answers is not in the document]
Comment (myihsan) [reply ios/Sources/MediaFeature/MediaScreen.swift:52]: I thought about it, but with the current implementation of `SwitchStore`, we have write something like below.
```swift
CaseLet(
    state: { (state: MediaState) in state == .needToInitialize ? () : nil },
    action: { (_: Never) -> MediaAction in },
    then: { _ in
        ProgressView()
            .onAppear { viewStore.send(.progressViewAppeared) }
    }
)
// Or
CaseLet(
    state: { (state: MediaState) in state == .needToInitialize ? () : nil },
    action: { (_: Void) -> MediaAction in .loadItems },
    then: {
        WithViewStore($0) { viewStore in
            ProgressView()
                .onAppear { viewStore.send(()) }
        }
    }
)
```
Is this okay for you?

[the remark this answers is not in the document]
Comment (ry-itto) [reply ios/Sources/MediaFeature/MediaScreen.swift:52]: This is not working?
In my environment, it seems to work correctly.
```swift
                CaseLet(
                    state: /MediaState.needToInitialize,
                    action: MediaAction.init(action:)) { _ in
                    ProgressView()
                        .onAppear { viewStore.send(.progressViewAppeared) }
                }
```

[the remark this answers is not in the document]
Comment (myihsan) [reply ios/Sources/MediaFeature/MediaScreen.swift]: Thanks! It works.
I have no idea why it broke when I tried it.

**Review by ry-itto — APPROVED** [review]
LGTM 💯 
Thanks 👍🏼

**Pull request merged**